library/x509write_crt.c

```diff
@@ -228,17 +228,17 @@ static int mbedtls_x509write_crt_set_key_identifier( mbedtls_x509write_cert
                                                       MBEDTLS_ASN1_CONSTRUCTED |
                                                       MBEDTLS_ASN1_SEQUENCE ) );
     }
-    return mbedtls_x509write_crt_set_extension(
-                   ctx,
-                   is_ca ? MBEDTLS_OID_AUTHORITY_KEY_IDENTIFIER :
-                               MBEDTLS_OID_SUBJECT_KEY_IDENTIFIER,
-                   is_ca ? MBEDTLS_OID_SIZE(
-                               MBEDTLS_OID_AUTHORITY_KEY_IDENTIFIER ) :
-                           MBEDTLS_OID_SIZE(
-                               MBEDTLS_OID_SUBJECT_KEY_IDENTIFIER ),
-                   0,
-                   buf + sizeof( buf ) - len,
-                   len );
+
+    if( is_ca )
+        return mbedtls_x509write_crt_set_extension( ctx,
+                MBEDTLS_OID_AUTHORITY_KEY_IDENTIFIER,
+                MBEDTLS_OID_SIZE( MBEDTLS_OID_AUTHORITY_KEY_IDENTIFIER ),
+                0, buf + sizeof(buf) - len, len );
+
+    return mbedtls_x509write_crt_set_extension( ctx,
+            MBEDTLS_OID_SUBJECT_KEY_IDENTIFIER,
+            MBEDTLS_OID_SIZE( MBEDTLS_OID_SUBJECT_KEY_IDENTIFIER ),
+            0, buf + sizeof(buf) - len, len );
```

Comment:
1. The control flow looks weird here: if is CA return, otherwise continue... but only to return something similar. `else` would make more sense.
1. It's easier to compare the two expressions if they have the same indentation.
1. Style: parentheses around `return`
```suggestion
    if( is_ca )
        return( mbedtls_x509write_crt_set_extension( ctx,
                MBEDTLS_OID_AUTHORITY_KEY_IDENTIFIER,
                MBEDTLS_OID_SIZE( MBEDTLS_OID_AUTHORITY_KEY_IDENTIFIER ),
                0, buf + sizeof(buf) - len, len ) );
    else
        return( mbedtls_x509write_crt_set_extension( ctx,
                MBEDTLS_OID_SUBJECT_KEY_IDENTIFIER,
                MBEDTLS_OID_SIZE( MBEDTLS_OID_SUBJECT_KEY_IDENTIFIER ),
                0, buf + sizeof(buf) - len, len ) );
```